plugin-api/src/main/java/org/hyperledger/besu/plugin/data/BadBlockCause.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Hyperledger Besu Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+package org.hyperledger.besu.plugin.data;
+
+/** Represents the reason a block is marked as "bad" */
+public interface BadBlockCause {
+
+  /**
+   * The reason why the block was categorized as bad
+   *
+   * @return The reason enum
+   */
+  BadBlockReason getReason();
+
+  /**
+   * A more descriptive explanation for why the block was marked bad
+   *
+   * @return the description
+   */
+  String getDescription();
+
+  /** An enum representing the reason why a block is marked bad */
+  enum BadBlockReason {
```

Comment:
@usmansaleem - I didn't completely follow why you had to move the nested enum in PR https://github.com/hyperledger/besu/pull/6716.  Tagging you here in case you think this is a problem?

Reply:
@mbaxter This should not be a problem in the interface as it will be treated as `static final` member. The reason for #6716 was weird errorprone issues where it was tripping and applied incorrect warnings (and only in that class).